gluster-block.spec.in

```diff
@@ -60,14 +60,18 @@ mv %{buildroot}/%{_initddir}/gluster-blockd.initd %{buildroot}/%{_initddir}/glus
 
 %if ( 0%{?_with_systemd:1} )
 %post
+%systemd_post gluster-block-target.service
 %systemd_post gluster-blockd.service
 
 %preun
+%systemd_preun gluster-block-target.service
 %systemd_preun gluster-blockd.service
-rm -rf %{_sharedstatedir}/gluster-block/gb_upgrade.status
 
 %postun
+%systemd_postun_with_restart gluster-block-target.service
 %systemd_postun_with_restart gluster-blockd.service
+if [ -e %{_sharedstatedir}/gluster-block/gb_upgrade.status ]; then
+    %{__rm} %{_sharedstatedir}/gluster-block/gb_upgrade.status
```

Comment:
Looks like if condition is not closed here ? 'fi' ?

Reply:
Fixed it.

If you do not like the %ghost version, maybe the following is better and will without any warning and error in rpmlint:
`
find %{_sharedstatedir}/gluster-block/ -name 'gb_upgrade.status' -delete
`

Reply:
@lxbsz with this changes do you still see the rpmlint warning about use of rm ?

Reply:
Yeah, it will be:

[gluster-block]# rpmlint build/rpmbuild/RPMS/x86_64/gluster-block-0.4-2.x86_64.rpm 
gluster-block.x86_64: W: no-version-in-last-changelog
gluster-block.x86_64: W: only-non-binary-in-usr-lib
gluster-block.x86_64: W: dangerous-command-in-%preun rm
1 packages and 0 specfiles checked; 0 errors, 3 warnings.
[gluster-block]#

Reply:
@lxbsz as we are sure what we are removing, this WARNING should be okay. Someone else can correct me.

If we look at glusterfs.spec there are 100's of rm usage.

Reply:
Yeah, but from my test, rpmlint seems will complain this rm warnings only in %preun and %postun, in glusterfs.spec it doesn't use any rm under them.

Reply:
strange for me. 

I'm okay, fixing this with 'find %{_sharedstatedir}/gluster-block/ -name 'gb_upgrade.status' -delete'
But technically, there is no difference.

Reply:
Yeah, true, technically there is no difference.